src/waypoint.jsx

```diff
@@ -295,8 +311,9 @@ export default class Waypoint extends React.Component {
 
   _getBounds() {
     const horizontal = this.props.horizontal;
-    const waypointTop = horizontal ? this._ref.getBoundingClientRect().left :
-      this._ref.getBoundingClientRect().top;
+    const { left, top, right, bottom } = this._ref.getBoundingClientRect();
+    const waypointTop = horizontal ? left : top;
+    const waypointBottom = horizontal ? right : bottom;
```

Comment:
I think that would require either a breaking change to the api (currently `waypointTop` is an exposed attribute) or inconsistent terminology between the internal codebase and the external api.